internal/lsp/server.go

```diff
@@ -288,6 +345,14 @@ func (s *Server) dispatchLoop(ctx context.Context) error {
 			}
 
 			handle := func() {
+				defer func() {
```

Comment:
Probably it's worth doing this on main; we really only want to kill the LS when a panic happens during the handling of a critical message, otherwise it should just be returning an error or something.

It's also possible to store off onto the LS the error and then in the test harness check after each use that no new errors appeared.